Cron/ProcessQueue.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace Algolia\AlgoliaSearch\Cron;
+
+use Algolia\AlgoliaSearch\Helper\ConfigHelper;
+use Algolia\AlgoliaSearch\Model\Queue;
+use Algolia\AlgoliaSearch\Service\AlgoliaCredentialsManager;
+
+class ProcessQueue
+{
+    public function __construct(
+        protected ConfigHelper $configHelper,
+        protected Queue $queue,
+        protected AlgoliaCredentialsManager $algoliaCredentialsManager
+    ) {}
+
+    public function execute()
+    {
+        if (!$this->configHelper->isQueueIndexerEnabled()) {
```

Comment:
I'm wondering if we should add another check here to bridge the transition from legacy extra crontab to using our new built-in cron? Maybe we could introduce a flag that says "use built-in cron" or something like that? The reason I think it's important is so people can upgrade without making any changes. (Otherwise they can end up with two running crons if they don't remember to remove the old entry from their server crontab.)  

It might make sense Initially to make it so that the "built in cron" is _disabled_ by default for 3.16 and _enabled_ by default for 3.17.

Reply:
That a totally good point ! Adding it.